src/compiler.ts

```diff
@@ -5925,6 +5933,11 @@ export class Compiler extends DiagnosticEmitter {
           );
           return module.unreachable();
         }
+
+        if (flow.actualFunction.is(CommonFlags.CONSTRUCTOR) && !(<Field>target).is(CommonFlags.INITIALIZED)) {
+          (<Field>target).set(CommonFlags.INITIALIZED);
+        }
```

Comment:
This looks like it needs tracking in flows, so it also covers branches like

```ts
constructor() {
  if (something) this.field = 1;
  // field might still not be initialized
}
```

Reply:
Definitely, I've marked that as one of the pending items for the PR to be ready for review :) Still wrapping my head around flows.

Reply:
haha, yeah. "flow" is quite tricky hybrid of cfg & dfg